velox/exec/tests/TableWriteTest.cpp

```diff
@@ -264,6 +294,115 @@ TEST_F(TableWriteTest, TestASecondCommitStrategy) {
       "SELECT * FROM tmp");
 }
 
+TEST_F(TableWriteTest, distinctPartitions) {
+  int32_t numPartitions = 50;
+  int32_t numBatches = 2;
+
+  auto rowType = ROW({"p0"}, {VARCHAR()});
+  std::vector<std::string> partitionNames = {"p0"};
+
+  auto inputFilePaths = makeFilePaths(numBatches);
+  std::vector<RowVectorPtr> vectors =
+      makeBatches(numBatches, [&](int32_t batch) {
+        return makeRowVector(
+            rowType->names(),
+            {makeFlatVector<StringView>(numPartitions, [&](auto row) {
+              return StringView(fmt::format("str_{}", row));
+            })});
+      });
+  for (int i = 0; i < numBatches; i++) {
+    writeToFile(inputFilePaths[i]->path, vectors[i]);
+  }
+
+  createDuckDbTable(vectors);
+
+  auto outputDirectory = TempDirectoryPath::create();
+  auto plan = PlanBuilder()
+                  .tableScan(rowType)
+                  .tableWrite(
+                      rowType->names(),
+                      std::make_shared<core::InsertTableHandle>(
+                          kHiveConnectorId,
+                          makeHiveInsertTableHandle(
+                              rowType->names(),
+                              rowType->children(),
+                              partitionNames,
+                              makeLocationHandle(outputDirectory->path))),
+                      CommitStrategy::kNoCommit,
+                      "rows")
+                  .project({"rows"})
+                  .planNode();
+
+  auto task = assertQuery(plan, inputFilePaths, "SELECT count(*) FROM tmp");
+
+  assertQuery(
+      PlanBuilder().tableScan(rowType).planNode(),
+      makeHiveConnectorSplits(outputDirectory),
+      "SELECT * FROM tmp");
+
+  EXPECT_EQ(
+      fileCount(outputDirectory), numPartitions * task->numOutputDrivers());
```

Comment:
What is the value of task->numOutputDrivers() here?

Reply:
`task->numOutputDrivers()` is just used to capture the # of drivers for the TableWriter. As you might already know, it's set here and now is 1.
https://github.com/facebookincubator/velox/blob/3bb9726871b3901f9fb37d61480f082a7c4892ee/velox/exec/LocalPlanner.cpp#L215-L221